### PR TITLE
Rework/pg category detail

### PR DESCRIPTION
**Pull request by erenkan**

### CODE REVIEW

File: scripts/components/FlEmptyItem.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import FlEmptyItemDesign from 'generated/my-components/FlEmptyItem';
+
+export default class FlEmptyItem extends FlEmptyItemDesign {
+    pageName?: string | undefined;
+    constructor(props?: any, pageName?: string) {
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+    get emptyTitle(): any {
```
Comment (alimoncul) [inline scripts/components/FlEmptyItem.ts]: string

File: scripts/components/FlEmptyItem.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return this.lblEmptyTitle.text;
+    }
+    set emptyTitle(value: any) {
```
Comment (alimoncul) [inline scripts/components/FlEmptyItem.ts]: string

File: scripts/components/FlEmptyItem.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        this.lblEmptyTitle.text = value;
+    }
+    get emptyImage(): any {
```
Comment (alimoncul) [inline scripts/components/FlEmptyItem.ts]: Image

File: scripts/components/FlEmptyItem.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        return this.imgEmpty.image;
+    }
+    set emptyImage(value: any) {
```
Comment (alimoncul) [inline scripts/components/FlEmptyItem.ts]: Image

File: scripts/components/LviEmptyItem.ts [hunk 5]
```diff
@@ -17,15 +17,15 @@ export default class LviEmptyItem extends LviEmptyItemDesign {
         return Screen.height / 1.5;
     }
     get emptyTitle(): any {
```
Comment (alimoncul) [inline scripts/components/LviEmptyItem.ts]: typings

File: scripts/pages/pgCategoryDetail.ts [hunk 6]
```diff
@@ -69,90 +72,93 @@ export default class PgCategoryDetail extends PgCategoryDetailDesign {
...
+                }, 500);
+            };
+        };
+    }
+    refreshGridView() {
+        this.gvProducts.itemCount = this.categoryProducts.length;
+        this.checkIfListEmpty();
+        this.gvProducts.refreshData();
+    }
+    checkIfListEmpty() {
```
Comment (alimoncul) [inline scripts/pages/pgCategoryDetail.ts:129]: just initialize the emptyItem once. Doing this on every refresh will be costly. Text assignment and visible dispatch can stay

File: scripts/pages/pgCategoryDetail.ts [hunk 7]
```diff
@@ -69,90 +72,93 @@ export default class PgCategoryDetail extends PgCategoryDetailDesign {
...
     superOnShow();
+    this.refreshGridView();
 }
 
 function onLoad(this: PgCategoryDetail, superOnLoad: () => void) {
     superOnLoad();
     this.headerBar.title = this.routeData.title;
-    // this.initSearchView();
-    console.log('onload route', this.routeData);
+    // this.initSearchView()
```
Comment (alimoncul) [inline scripts/pages/pgCategoryDetail.ts]: remove comment

Comment (erenkan) [reply scripts/pages/pgCategoryDetail.ts]: OK.

File: scripts/components/LviEmptyItem.ts (hunk 5 above)
Comment (erenkan) [reply scripts/components/LviEmptyItem.ts]: OK.

File: scripts/components/FlEmptyItem.ts (hunk 4 above)
Comment (erenkan) [reply scripts/components/FlEmptyItem.ts]: OK.

File: scripts/components/FlEmptyItem.ts (hunk 3 above)
Comment (erenkan) [reply scripts/components/FlEmptyItem.ts]: OK.

File: scripts/components/FlEmptyItem.ts (hunk 2 above)
Comment (erenkan) [reply scripts/components/FlEmptyItem.ts]: OK.

File: scripts/components/FlEmptyItem.ts (hunk 1 above)
Comment (erenkan) [reply scripts/components/FlEmptyItem.ts]: OK.

File: scripts/pages/pgCategoryDetail.ts (hunk 6 above)
Comment (erenkan) [reply scripts/pages/pgCategoryDetail.ts:129]: OK.

**Pull request merged**